### PR TITLE
Add toggles

### PR DESCRIPTION
**Pull request by JakobGray**
Issue: https://github.com/stolostron/backlog/issues/20010

Signed-off-by: Jakob Gray <20209054+JakobGray@users.noreply.github.com>

### CODE REVIEW

**Review by zkayyali812 — COMMENTED** [review]
Did you mean to deprecate other fields in our spec which control the toggling of our components?
Like - 
- EnableClusterProxyAddon
- DisableUpdateClusterImageSets?
- Do we want to move `Ingress` under the ingress componentConfig?

**Comment by JakobGray** [comment]
> Did you mean to deprecate other fields in our spec which control the toggling of our components? Like -
> 
> * EnableClusterProxyAddon
> * DisableUpdateClusterImageSets?
> * Do we want to move `Ingress` under the ingress componentConfig?

You're right EnableClusterProxyAddon should be deprecated as that will be replaced by the toggle. Those other two I don't plan on changing as they are out of scope for this story

File: controllers/common.go [hunk 1]
```diff
@@ -928,10 +976,10 @@ func (r *MultiClusterHubReconciler) ensureMultiClusterEngine(multiClusterHub *op
 		r.Log.Info("Updating preexisting MCE")
 		existingMCE.Spec.AvailabilityConfig = mcev1.AvailabilityType(multiClusterHub.Spec.AvailabilityConfig)
 		existingMCE.Spec.NodeSelector = multiClusterHub.Spec.NodeSelector
-		if multiClusterHub.ComponentEnabled(operatorv1.ManagedServiceAccount) {
-			existingMCE.Spec.ComponentConfig = multiclusterengine.GetComponentConfig(multiClusterHub)
-		}
-		existingMCE.Spec.ComponentConfig = multiclusterengine.GetComponentConfig(multiClusterHub)
+		// if multiClusterHub.Enabled(operatorv1.ManagedServiceAccount) {
```
Comment (zkayyali812) [inline controllers/common.go:979]: Can these be removed?

File: pkg/multiclusterengine/multiclusterengine.go [hunk 2]
```diff
@@ -54,9 +54,9 @@ func MultiClusterEngine(m *operatorsv1.MultiClusterHub) *mcev1.MultiClusterEngin
 			NodeSelector:    m.Spec.NodeSelector,
 		},
 	}
-	if m.ComponentEnabled(operatorsv1.ManagedServiceAccount) {
-		mce.Spec.ComponentConfig = GetComponentConfig(m)
-	}
+	// if m.Enabled(operatorsv1.ManagedServiceAccount) {
```
Comment (zkayyali812) [inline pkg/multiclusterengine/multiclusterengine.go:57]: same

File: pkg/multiclusterengine/multiclusterengine.go [hunk 3]
```diff
@@ -71,14 +71,14 @@ func GetSupportedAnnotations(m *operatorsv1.MultiClusterHub) map[string]string {
...
 
-func GetComponentConfig(m *operatorsv1.MultiClusterHub) *mcev1.ComponentConfig {
-	componentConfig := &mcev1.ComponentConfig{
-		ManagedServiceAccount: &mcev1.ManagedServiceAccountConfig{
-			Enable: m.ComponentEnabled(operatorsv1.ManagedServiceAccount),
-		},
-	}
-	return componentConfig
-}
+// func GetComponentConfig(m *operatorsv1.MultiClusterHub) *mcev1.ComponentConfig {
```
Comment (zkayyali812) [inline pkg/multiclusterengine/multiclusterengine.go:74]: same

File: controllers/common.go (hunk 1 above)
Comment (JakobGray) [reply controllers/common.go]: Removed

File: pkg/multiclusterengine/multiclusterengine.go (hunk 2 above)
Comment (JakobGray) [reply pkg/multiclusterengine/multiclusterengine.go]: Removed

File: pkg/multiclusterengine/multiclusterengine.go (hunk 3 above)
Comment (JakobGray) [reply pkg/multiclusterengine/multiclusterengine.go]: Removed

**Review by zkayyali812 — APPROVED** [review]

**Pull request merged**